### PR TITLE
Price filter Graduation/Wedding/Anniversary/Christmas pages

### PR DESCRIPTION
**Pull request by alocasia28**
Component added to all pages and logic updated to enable filtering by price and category. 
Added rendering of error message when no gifts found in the filtered range. 

Error message: 
Screenshot 2024-04-07 at 12 29 30 pm

Screenshot 2024-04-07 at 12 11 50 pm

Screenshot 2024-04-07 at 12 12 07 pm

Screenshot 2024-04-07 at 12 12 23 pm

Screenshot 2024-04-07 at 12 12 23 pm

### CODE REVIEW

File: src/components/PriceFilter.jsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ function PriceFilter({ priceFilter = { min: 0, max: 1000 }, updatePriceFilter, c
                         {" "}
                         Above $250{" "}
                     </button>
-                    <button className="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded" onClick={clearPriceFilter}>Clear Filter</button>
+                    <button className="border rounded-md bg-white border-primary-600 text-primary-300 font-bold py-2 px-4 rounded hover:bg-primary-300 hover:text-white py-2 px-4" onClick={clearPriceFilter}>Clear Filter</button>
```
Comment (RobyOneJ) [inline src/components/PriceFilter.jsx:65]: 1. text bold is not in line with any other style - should that be removed?

2. also should the 'clear' button be disabled if not price selection was made?

Comment (alocasia28) [reply src/components/PriceFilter.jsx:65]: 1. I added it in to differentiate, but Anastasia is going to come in and fix styling for these. 
2. I am going to leave it as is for now so we can get on with other things. Given it doesn't do anything when no filters are active, it's not as bad as the apply filter one where it was breaking the gifts rendering

Comment (alocasia28) [reply src/components/PriceFilter.jsx:65]: Updated typo in header.

Comment (RobyOneJ) [reply src/components/PriceFilter.jsx:65]: 1. if Anastasiia is going to address button style - I accept it.
2. not a nice user experience but true that functionality wise is not breaking like the other one 
3. thanks for fixing that typo

**Comment by RobyOneJ** [comment]
Issue with footer (as shown in screenshot in the PR description) doesn't look like is related to this PR so will approve it and Sandra's PR on styling, not merged yet, is going to fix it.

**Review by RobyOneJ — APPROVED** [review]

**Pull request merged**